### PR TITLE
[SCOTCH] Version 7.0.3

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Comment by amontoison** [comment]
@giordano I will compile the last release of SCOTCH 6.1.x before, can I do multiple folders (SCOTCH@6, SCOTCH@7) like MUMPS or OpenBLAS to keep both `build_tarball.jl` ?

**Comment by ViralBShah** [comment]
Yes fine to do multiple versions if we need both simultaneously.

**Comment by amontoison** [comment]
I don't know how to solve the error
```shell
/workspace/srcdir/scotch-v7.0.3/src/libscotch/common_thread_system.c:118:24: error: use of undeclared identifier '_SC_NPROCESSORS_ONLN'
    corenbr = sysconf (_SC_NPROCESSORS_ONLN);
```
on the FreeBSD platform.
I propose to drop it.

File: S/SCOTCH/build_tarballs.jl [hunk 1]
```diff
@@ -3,49 +3,89 @@
...
+    LibraryProduct("libscotchmetisv3", :libscotchmetisv3),
+    LibraryProduct("libscotchmetisv5", :libscotchmetisv5)
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a"))
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a")),
+    Dependency(PackageSpec(name="Bzip2_jll", uuid="6e34b625-4abd-537c-b88f-471c36dfa7a0")),
```
Comment (giordano) [inline S/SCOTCH/build_tarballs.jl:87]:
```suggestion
    Dependency(PackageSpec(name="Bzip2_jll", uuid="6e34b625-4abd-537c-b88f-471c36dfa7a0"); compat="1.0.8"),
```

File: S/SCOTCH/build_tarballs.jl [hunk 2]
```diff
@@ -3,49 +3,89 @@
...
 
 name = "SCOTCH"
-version = v"6.1.3"
+version = v"7.0.3"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://gitlab.inria.fr/scotch/scotch/-/archive/v$(version)/scotch-v$(version).tar.gz","4e54f056199e6c23d46581d448fcfe2285987e5554a0aa527f7931684ef2809e"),
-    DirectorySource("./bundled")
+    ArchiveSource("https://gitlab.inria.fr/scotch/scotch/-/archive/v$(version)/scotch-v$(version).tar.gz", "5b5351f0ffd6fcae9ae7eafeccaa5a25602845b9ffd1afb104db932dd4d4f3c5"),
```
[the remark this answers is not in the document]
Comment (amontoison) [reply S/SCOTCH/build_tarballs.jl]: I replaced the `ArchiveSource` by a `GitSource`.

**Pull request merged**